packages/wrapped-keys/src/lib/litActions.ts

```diff
@@ -1,146 +1,193 @@
-            { waitForResponse: true, name: 'gasLimit' },
-            async () => {
-                const rpcUrl = await Lit.Actions.getRpcUrl({ chain: unsignedTransaction.chain });
-                const provider = new ethers.providers.JsonRpcProvider(rpcUrl);
-                
-                try {
-                    const gasLimit = await provider.estimateGas(tx);
-                    return ethers.utils.hexlify(gasLimit);
-                } catch (err) {
-                    const errorMessage = 'Error: When estimating gas- ' + err.message;
-                    Lit.Actions.setResponse({ response: errorMessage });
-                    return;
-                }
-            }
-        )
-    }
+  if (!unsignedTransaction.chain) {
+    Lit.Actions.setResponse({
+      response: 'Error: Missing required field: chain',
+    });
+    return;
+  }
 
-    console.log('tx');
-    console.log(tx);
+  if (!unsignedTransaction.value) {
+    Lit.Actions.setResponse({
+      response: 'Error: Missing required field: value',
+    });
+    return;
+  }
 
-    const signedTx = await Lit.Actions.runOnce(
-        { waitForResponse: true, name: "signedTx" },
-        async () => {
-            try {
-                return wallet.signTransaction(tx);
-            } catch (error) {
-                const errorMessage = 'Error: When signing transaction- ' + err.message;
-                Lit.Actions.setResponse({ response: errorMessage });
-                return;
-            }
+  if (!unsignedTransaction.chainId) {
+    Lit.Actions.setResponse({
+      response: 'Error: Missing required field: chainId',
+    });
+    return;
+  }
+
+  const decryptedPrivateKey = await Lit.Actions.decryptToSingleNode({
+    accessControlConditions,
+    ciphertext,
+    dataToEncryptHash,
+    chain: 'ethereum',
+    authSig: null,
+  });
+
+  console.log('decryptedPrivateKey');
+  console.log(decryptedPrivateKey);
+
+  if (!decryptedPrivateKey) {
+    // Exit the nodes which don't have the decryptedData
+    return;
+  }
+
+  const privateKey = decryptedPrivateKey.startsWith(LIT_PREFIX)
+    ? decryptedPrivateKey.slice(LIT_PREFIX.length)
+    : decryptedPrivateKey;
+  const wallet = new ethers.Wallet(privateKey);
+
+  console.log('unsignedTransaction.chain', unsignedTransaction.chain);
+  console.log('pkpAddress', pkpAddress);
+  const nonce = await Lit.Actions.getLatestNonce({
+    address: wallet.address,
+    chain: unsignedTransaction.chain,
+  });
+  console.log('nonce');
+  console.log(nonce);
+
+  const tx = {
+    to: unsignedTransaction.toAddress,
+    value: ethers.utils.hexlify(
+      ethers.utils.parseEther(unsignedTransaction.value)
+    ),
+    chainId: unsignedTransaction.chainId,
+    data: unsignedTransaction.dataHex,
+    nonce,
+  };
+
+  if (unsignedTransaction.gasPrice) {
+    tx.gasPrice = ethers.utils.parseUnits(unsignedTransaction.gasPrice, 'gwei');
+  } else {
+    tx.gasPrice = await Lit.Actions.runOnce(
```

Comment:
This is not need as we already exit if the `decryptedPrivateKey` is undefined since we ran the function `decryptToSingleNode` above. So this point all this is being run only on a single node